### ISSUE TITLE
jupyterhub/tornado unable to find templates, `style.min.css` and `require.js`

### ISSUE THREAD

**Issue by psteinb** [jupyterhub/jupyterhub#1341]
Hi -

- Please use the [GitHub issue](https://github.com/jupyterhub/jupyterhub/issues)
  search feature to check if your issue has been asked already. If it has,
  please add your comments to the existing issue.

I already did that, but found nothing related.

**How to reproduce the issue**

I wanted to setup jupyterhub on our HPC infrastructure. People have done so and reported very good experiences across the web. I installed jupyterhub like so (as myself not `root`):
```
$ wget https://github.com/jupyterhub/jupyterhub/archive/0.7.2.tar.gz
$ tar xf 0.7.2.tar.gz
$ cd jupyterhub-0.7.2
$ pip3 install --prefix /home/courses/sw/apps/jupyterhub/0.7.2 .
```
All prerequisites are installed accordingly and the environment is changed to point `PYTHONPATH` and `PATH` to the prefix mentioned above. I then switch to being root and run:
```
# jupyterhub --debug -f /home/courses/sw/etc/jupyterhub/jupyterhub_config.py --log-file=/home/courses/sw/etc/jupyterhub/log
```

**What you expected to happen**

jupyter works as expected.

**What actually happens**

1) jupyter hub crashed as it couldn't find the template directory. I appended `/home/courses/sw/apps/jupyterhub/0.7.2/share/jupyter/hub/templates` to `c.JupyterHub.template_path` in the config and I was able to use jupyterhub through the Web UI. But the UI looks awkward.

2) during the first HTTP GET to the hub landing page, I see the following in the log:
```
[I 2017-08-16 13:11:45.709 JupyterHub app:1485] JupyterHub is now running at http://127.0.0.1:4043/
[E 2017-08-16 13:11:46.756 JupyterHub web:2653] Could not open static file '/root/static/css/style.min.css'
[E 2017-08-16 13:11:46.757 JupyterHub web:2653] Could not open static file '/root/static/components/requirejs/require.js'
[I 2017-08-16 13:11:46.760 JupyterHub log:100] 200 GET /hub/spawn (steinbac@10.1.251.33) 26.39ms
[W 2017-08-16 13:11:46.790 JupyterHub log:100] 404 GET /hub/static/css/style.min.css (@10.1.251.33) 0.46ms
[W 2017-08-16 13:11:46.791 JupyterHub log:100] 404 GET /hub/static/components/requirejs/require.js (@10.1.251.33) 0.41ms
[W 2017-08-16 13:11:46.814 JupyterHub log:100] 404 GET /hub/logo (@10.1.251.33) 0.71ms
```

**Share what version of JupyterHub you are using**
```
# jupyterhub --version
0.7.2
# configurable-http-proxy --version
2.0.4
# jupyterhub troubleshoot
[I 2017-08-16 13:12:35.546 JupyterHub app:724] Loading cookie_secret from /root/jupyterhub_cookie_secret
[W 2017-08-16 13:12:35.588 JupyterHub app:365] 
    Generating CONFIGPROXY_AUTH_TOKEN. Restarting the Hub will require restarting the proxy.
    Set CONFIGPROXY_AUTH_TOKEN env or JupyterHub.proxy_auth_token config to avoid this message.
    
[W 2017-08-16 13:12:35.591 JupyterHub app:864] No admin users, admin interface will be unavailable.
[W 2017-08-16 13:12:35.592 JupyterHub app:865] Add any administrative users to `c.Authenticator.admin_users` in config.
[I 2017-08-16 13:12:35.592 JupyterHub app:892] Not using whitelist. Any authenticated user will be allowed.
[I 2017-08-16 13:12:35.604 JupyterHub app:1453] Hub API listening on http://127.0.0.1:8081/hub/
[W 2017-08-16 13:12:35.606 JupyterHub app:1174] Running JupyterHub without SSL.  I hope there is SSL termination happening somewhere else...
[I 2017-08-16 13:12:35.606 JupyterHub app:1176] Starting proxy @ http://*:8000/
13:12:35.811 - info: [ConfigProxy] Adding route / -> http://127.0.0.1:8081
13:12:35.818 - info: [ConfigProxy] Proxying http://*:8000 to http://127.0.0.1:8081
13:12:35.819 - info: [ConfigProxy] Proxy API at http://127.0.0.1:8001/api/routes
[I 2017-08-16 13:12:35.914 JupyterHub app:1485] JupyterHub is now running at http://127.0.0.1:8000/
```
AS I want to use Batchspawner in conjunction with Wrapspawner, I'd love to get these fixed. It looks to me that the part resolved by tornado to read in the html to render is quite off. According to the debug messages, jupyterhub/tornado keeps looking inside /root for `require.js`, `style.min.css` and friends. 

Any help would be highly appreciated.

**Comment by psteinb** [comment]
just ran jupyterhub as myself on the same machine, the symptoms are identical. Here is my condensed `jupyterhub_config.py`:

```
c.JupyterHub.db_url = 'sqlite:////tmp/jupyterhub.sqlite'
c.JupyterHub.ip = '0.0.0.0'
c.JupyterHub.port = 4043
c.JupyterHub.hub_port = 8081
c.JupyterHub.hub_ip = '0.0.0.0'
c.JupyterHub.cookie_max_age_days = 8
c = get_config()
c.JupyterHub.spawner_class = 'batchspawner.SlurmSpawner'
c.Spawner.http_timeout = 120
c.SlurmSpawner.batch_script = '''#!/bin/bash
#SBATCH ...
{cmd}
'''
c.JupyterHub.spawner_class = 'wrapspawner.ProfilesSpawner'
c.Spawner.http_timeout = 120
c.BatchSpawnerBase.req_nprocs = '2'
c.BatchSpawnerBase.req_runtime = '12:00:00'
c.ProfilesSpawner.profiles = [
   ( "Local server", 'local', 'jupyterhub.spawner.LocalProcessSpawner', {'ip':'0.0.0.0'} ),
   ('Interactive GPU - 6 cores,  16 GB, 8 hours',
    'interactive-gpu',
    'batchspawner.SlurmSpawner',
      dict(req_nprocs='2', req_queue='gpu',
          req_runtime='8:00:00', req_memory='16gb')
   )
   ]
c.JupyterHub.template_paths = ["/home/courses/sw/apps/jupyterhub/0.7.2/share/jupyter/hub/templates"]
c.Authenticator.admin_users = set()
c.Authenticator.whitelist = set(['xxx','yyy'])
```

**Comment by minrk** [comment]
Instead of installing from the GitHub archive, which is generated automatically by GitHub from the git repo and *not* the published source distributions we publish, try installing from the standard PyPI sdist:

    pip uninstall jupyterhub
    pip install jupyterhub==0.7.2

This should ensure that you get everything without needing to specify any `.template_paths` configuration.

**Comment by psteinb** [comment]
alright, I'll try that and report back. Thanks for the quick response.

**Comment by psteinb** [comment]
just did a

```
pip3 install --prefix=/path/to/0.8.0b1 jupyterhub
```

and the symptoms are the same. Should I try `pip` instead?

**Comment by willingc** [comment]
If you install without setting `PYTHONPATH` and using `--prefix` do you see the same errors?

For released JupyterHub 0.7.2:
```
python3 -m pip uninstall jupyterhub
python3 -m pip install jupyterhub==0.7.2
```

If you wish to use the beta version:
```
python3 -m pip install --pre -U jupyterhub
```

**Comment by psteinb** [comment]
alright, I got it. When we do installations on our cluster, we typically do:

```
$ pip install --prefix=/path/to/apps/myapp/version myapp
$ cd /path/to/apps/myapp/
$ ln -s version current
```
we then symlink the contents of /path/to/apps/myapp/current/bin into some central folder /sw/bin. this way we only have to change a symlink when new versions come in. I did the same with jupyterhub and that confused jupyterhub because this is what ended up in `PATH`:
`/sw/bin` which contains a symlink to the actual jupyterhub app. I am guessing that jupyterhub internally tries to find out where it is located, right?
I just changed PATH and PYTHONPATH to point to the exact location of jupyterhub, namely `/path/to/apps/jupyterhub/0.7.2` and it works flawlessly. Note, this 0.7.2 was installed from a github clone. Issue solved.
